Octokit.Tests.Integration/Clients/RepositoryBranchesClientTests.cs

```diff
@@ -543,4 +544,999 @@ public TheDeleteBranchProtectionMethod()
+    {
+        IRepositoryBranchesClient _client;
+        OrganizationRepositoryWithTeamContext _orgRepoContext;
+
+        public TheGetProtectedBranchTeamRestrictionsMethod()
+        {
+            var github = Helper.GetAuthenticatedClient();
+            _client = github.Repository.Branch;
+
+            _orgRepoContext = github.CreateOrganizationRepositoryWithProtectedBranch().Result;
+        }
+
+        [IntegrationTest]
+        public async Task GetsProtectedBranchTeamRestrictionsForOrgRepo()
+        {
+            var repoOwner = _orgRepoContext.RepositoryContext.RepositoryOwner;
+            var repoName = _orgRepoContext.RepositoryContext.RepositoryName;
+            var restrictions = await _client.GetProtectedBranchTeamRestrictions(repoOwner, repoName, "master");
+
+            Assert.NotNull(restrictions);
+            Assert.Equal(1, restrictions.Count);
+        }
+
+        [IntegrationTest]
+        public async Task GetsProtectedBranchTeamRestrictionsForOrgRepoWithRepositoryId()
+        {
+            var repoId = _orgRepoContext.RepositoryContext.RepositoryId;
+            var restrictions = await _client.GetProtectedBranchTeamRestrictions(repoId, "master");
+
+            Assert.NotNull(restrictions);
+            Assert.Equal(1, restrictions.Count);
+        }
+
+        public void Dispose()
+        {
+            if (_orgRepoContext != null)
+                _orgRepoContext.Dispose();
+        }
+    }
+
+    public class TheSetProtectedBranchTeamRestrictionsMethod : IDisposable
+    {
+        IRepositoryBranchesClient _client;
+        OrganizationRepositoryWithTeamContext _orgRepoContext;
+        IGitHubClient _github;
+
+        public TheSetProtectedBranchTeamRestrictionsMethod()
+        {
+            _github = Helper.GetAuthenticatedClient();
+            _client = _github.Repository.Branch;
+
+            _orgRepoContext = _github.CreateOrganizationRepositoryWithProtectedBranch().Result;
+        }
+
+        [IntegrationTest]
+        public async Task SetsProtectedBranchTeamRestrictionsForOrgRepo()
+        {
+            var repoOwner = _orgRepoContext.RepositoryContext.RepositoryOwner;
+            var repoName = _orgRepoContext.RepositoryContext.RepositoryName;
+
+            var contextOrgTeam2 = await _github.CreateTeamContext(Helper.Organization, new NewTeam(Helper.MakeNameWithTimestamp("team2")));
+
+            // Grant team push access to repo
+            await _github.Organization.Team.AddRepository(
+                contextOrgTeam2.TeamId,
+                repoOwner,
+                repoName,
+                new RepositoryPermissionRequest(Permission.Push));
+
+            var newTeam = new BranchProtectionTeamCollection() { contextOrgTeam2.TeamName };
+            var restrictions = await _client.SetProtectedBranchTeamRestrictions(repoOwner, repoName, "master", newTeam);
+
+            Assert.NotNull(restrictions);
+            Assert.Equal(1, restrictions.Count);
```

Comment:
The branch protection team restrictions started off with 1 team in them, and this test (hopefully) change the protection to team2 instead.  but just asserting the count of team restrictions is 1 doesnt really prove that the update worked (what if it didnt work and the original team is the one that was returned?).  So perhaps assert on the team's name rather than just the count?